tests/aviftest.c

```diff
@@ -270,18 +270,19 @@ static avifResult avifIOTestReaderRead(struct avifIO * io, uint32_t readFlags, u
     avifIOTestReader * reader = (avifIOTestReader *)io;
 
     // Sanitize/clamp incoming request
-    if (offset >= reader->rodata.size) {
-        offset = 0;
-        size = 0;
+    if (offset > reader->rodata.size) {
+        // The offset is past the end of the buffer.
+        return AVIF_RESULT_IO_ERROR;
     }
-    if ((offset + size) > reader->rodata.size) {
-        size = reader->rodata.size - offset;
+    uint64_t availableSize = reader->rodata.size - offset;
+    if (size > availableSize) {
+        size = availableSize;
     }
 
-    if (offset >= reader->availableBytes) {
+    if (offset > reader->availableBytes) {
```

Comment:
Note that if offset == reader->availableBytes && size == 0, the new code will return AVIF_RESULT_OK.  I made this change so that this test looks like the test at line 273. I believe the function's behavior in this corner case doesn't really matter.